tests/util/mock_app.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 PingCAP, Inc. Licensed under Apache-2.0.
+
+package util
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/fx"
+	"go.uber.org/fx/fxtest"
+
+	"github.com/pingcap/tidb-dashboard/pkg/apiserver"
+	"github.com/pingcap/tidb-dashboard/pkg/config"
+	"github.com/pingcap/tidb-dashboard/util/featureflag"
+)
+
+type App struct {
+	*fxtest.App
+
+	tb fxtest.TB
+}
+
+func NewMockApp(tb fxtest.TB, tidbVersion string, c *config.Config, opts ...fx.Option) *App {
+	allOpts := make([]fx.Option, 0, len(opts)+1)
+	allOpts = append(allOpts,
+		apiserver.Modules,
+		fx.Supply(featureflag.NewRegistry(tidbVersion)),
```

Comment:
This seems tricky. Currently we remove it from the apiserver.Modules and then supply it later everywhere. It would be better if it behaves as "overwrite" instead.